Time - Denisse

```diff
@@ -0,0 +1,83 @@
+# Program that will calculate and display information  
+# regarding the trips/rides all drivers have completed.
+
+# Hold the hard coded information regarding each trip
+drivers_stats = {
+  DR0001: [
+    {date: "3rd Feb 2016", cost: 10, rider_id: "RD0003", rating: 3},
+    {date: "3rd Feb 2016", cost: 30, rider_id: "RD0015", rating: 4},
+    {date: "5th Feb 2016", cost: 45, rider_id: "RD0003", rating: 2}
+  ],
+  DR0002: [
+    {date: "3rd Feb 2016", cost: 25, rider_id: "RD0073", rating: 5},
+    {date: "4th Feb 2016", cost: 15, rider_id: "RD0013", rating: 1},
+    {date: "5th Feb 2016", cost: 35, rider_id: "RD0066", rating: 3}
+  ],
+  DR0003: [
+    {date: "4th Feb 2016", cost: 5, rider_id: "RD0066", rating: 5},
+    {date: "5th Feb 2016", cost: 50, rider_id: "RD0003", rating: 2}
+  ],
+  DR0004: [
+    {date: "3rd Feb 2016", cost: 5, rider_id: "RD0022", rating: 5},
+    {date: "4th Feb 2016", cost: 10, rider_id: "RD0022", rating: 4},
+    {date: "5th Feb 2016", cost: 20, rider_id: "RD0073", rating: 5}
+  ]  
+}
+
+# Arrays to hold trip's information
+trip_cost = []
+trip_rating = []
+total_trip_cost = []
+trip_ratings = []
+
+# Gather all information per ride, store in designated arrays and display results as needed
+puts "\nThe results are as follow:"
+
+# Find all trip/ride cost and rating per driver
+drivers_stats.each do |driver_id, rides|
+  rides.each do |ride_info|
+    trip_cost << ride_info[:cost]
+    trip_rating << ride_info[:rating]
+  end
+
+  # Average the trip's rating and store for later use
+  trip_ratings << '%.2f' % ((trip_rating.sum.to_f/trip_rating.length))
+  # Sum all trip cost per driver and store for later use
+  total_trip_cost << (trip_cost.sum)
+  
+  # Display number of rides done for driver
+  puts "\nDriver #{driver_id} had a total of #{drivers_stats[driver_id].length} rides"
+  # Display earnings per trip
+  puts "Making a grand total of $#{trip_cost.sum} for all the awesome rides"
+  # Calculate and display the average rating for driver
+  puts "Their average rating is #{'%.2f' % (trip_rating.sum.to_f/trip_rating.length)}"
+  
+  # Clear the arrays for next driver
+  trip_cost.clear
+  trip_rating.clear
+end 
+
+# Calculate and display driver with the highest average rating
+puts "\nThe driver with the highest average rating was:"
+puts "Driver DR000#{trip_ratings.index(trip_ratings.max) + 1} with an average rating of #{trip_ratings.max}" 
```

Comment:
This is a clever way of using the index! Also, consider how you would determine the driver that corresponded to the max trip rating if the drivers were not have numerically continuous IDs.